apps/whale-api/src/module.api/poolpair.fees.service.e2e.ts

```diff
@@ -0,0 +1,510 @@
+    await testing.generate(1)
+  }
+
+  await addPoolLiquidity(container, {
+    tokenA: 'DUSD',
+    amountA: 10,
+    tokenB: 'DFI',
+    amountB: 10,
+    shareAddress: await getNewAddress(container)
+  })
+
+  await testing.generate(1)
+
+  await addPoolLiquidity(container, {
+    tokenA: 'CAT',
+    amountA: 100,
+    tokenB: 'DFI',
+    amountB: 1000,
+    shareAddress: await getNewAddress(container)
+  })
+  await testing.generate(1)
+
+  await addPoolLiquidity(container, {
+    tokenA: 'DOG',
+    amountA: 10,
+    tokenB: 'DFI',
+    amountB: 100,
+    shareAddress: await getNewAddress(container)
+  })
+  await testing.generate(1)
+
+  await addPoolLiquidity(container, {
+    tokenA: 'KOALA',
+    amountA: 10,
+    tokenB: 'DFI',
+    amountB: 100,
+    shareAddress: await getNewAddress(container)
+  })
+  await testing.generate(1)
+
+  await addPoolLiquidity(container, {
+    tokenA: 'FISH',
+    amountA: 5,
+    tokenB: 'DFI',
+    amountB: 100,
+    shareAddress: await getNewAddress(container)
+  })
+  await testing.generate(1)
+
+  await addPoolLiquidity(container, {
+    tokenA: 'TURTLE',
+    amountA: 1,
+    tokenB: 'DFI',
+    amountB: 100,
+    shareAddress: await getNewAddress(container)
+  })
+  await testing.generate(1)
+
+  await addPoolLiquidity(container, {
+    tokenA: 'PANDA',
+    amountA: 2,
+    tokenB: 'DFI',
+    amountB: 100,
+    shareAddress: await getNewAddress(container)
+  })
+  await testing.generate(1)
+
+  await addPoolLiquidity(container, {
+    tokenA: 'RABBIT',
+    amountA: 7,
+    tokenB: 'DFI',
+    amountB: 100,
+    shareAddress: await getNewAddress(container)
+  })
+  await testing.generate(1)
+
+  // dex fee set up
+  await container.call('setgov', [{
+    ATTRIBUTES: {
+      'v0/poolpairs/4/token_a_fee_pct': '0.001', // CAT
+      'v0/poolpairs/4/token_a_fee_direction': 'in', // CAT
+      'v0/poolpairs/4/token_b_fee_pct': '0.002', // DFI
+      'v0/poolpairs/4/token_b_fee_direction': 'in', // DFI
+
+      'v0/poolpairs/6/token_a_fee_pct': '0.003', // DOG
+      'v0/poolpairs/6/token_a_fee_direction': 'out', // DOG
+      'v0/poolpairs/6/token_b_fee_pct': '0.004', // DFI
+      'v0/poolpairs/6/token_b_fee_direction': 'out', // DFI
+
+      'v0/poolpairs/8/token_a_fee_pct': '0.005', // KOALA
+      'v0/poolpairs/8/token_a_fee_direction': 'in', // KOALA
+
+      'v0/poolpairs/10/token_b_fee_pct': '0.006', // FISH
+      'v0/poolpairs/10/token_b_fee_direction': 'out', // FISH
+
+      'v0/poolpairs/12/token_a_fee_pct': '0.007', // TURTLE
+      'v0/poolpairs/12/token_a_fee_direction': 'both', // TURTLE
+
+      'v0/poolpairs/14/token_b_fee_pct': '0.008', // PANDA
+      'v0/poolpairs/14/token_b_fee_direction': 'both', // PANDA
+
+      'v0/poolpairs/16/token_a_fee_pct': '0.009', // RABBIT
+      'v0/poolpairs/16/token_a_fee_direction': 'both', // RABBIT
+      'v0/poolpairs/16/token_b_fee_pct': '0.010', // RABBIT
+      'v0/poolpairs/16/token_b_fee_direction': 'both' // RABBIT
```

Comment:
add another pair and a test where only `token_a_fee_pct`, `token_b_fee_pct` is set.
related to https://github.com/JellyfishSDK/jellyfish/pull/1601/files#r919629537

Reply:
[Here](https://github.com/JellyfishSDK/jellyfish/pull/1601/commits/db69caf13d74c127498d9e663cac9f6b36911eee#diff-d0967f195ebd61af3cfd8c80b467983958ae965caa4857a2d283b4095b6cb07eR498), good catch, @surangap! 🙏